compiler/rustc_target/src/spec/base/apple/mod.rs

```diff
@@ -151,7 +156,7 @@ pub fn opts(os: &'static str, arch: Arch, abi: TargetAbi) -> TargetOptions {
         // macOS has -dead_strip, which doesn't rely on function_sections
         function_sections: false,
         dynamic_linking: true,
-        pre_link_args: pre_link_args(os, arch, abi),
+        pre_link_args,
```

Comment:
Why was this moved outside of `opts`?
I think we can still create lazy args here (especially if `fn pre_link_args` already returns lazy in the first place).

Reply:
`fn pre_link_args` now returns lazy, this is no longer the case.